environment-pipelines/buildspec-apply.yml

```diff
@@ -17,6 +17,32 @@ phases:
       - echo -e "\nWorking on environment ${ENVIRONMENT}"
       - cd "terraform/environments/${ENVIRONMENT}"
       - terraform apply plan.tfplan
+      - echo -e "\nGenerating manifests and deploying AWS Copilot environment resources"
+      - |
+          if [[ "${COPILOT_PROFILE}" != *"prod"* ]]
+          then
+            cd "${CODEBUILD_SRC_DIR}"
+            platform-helper environment generate --name "${ENVIRONMENT}"
+            copilot env init --name "${ENVIRONMENT}" --profile "${COPILOT_PROFILE}" --default-config
+            platform-helper copilot make-addons
+            copilot env deploy --name "${ENVIRONMENT}"
+          else
+            assumed_role=$(aws sts assume-role --role-arn "${TRIGGERING_ACCOUNT_CODEBUILD_ROLE}" --role-session-name "trigger-copilot-env-deploy-$(date +%s)")    
+            NON_PROD_AWS_ACCESS_KEY_ID=$(echo $assumed_role | jq -r .Credentials.AccessKeyId)
+            NON_PROD_AWS_SECRET_ACCESS_KEY=$(echo $assumed_role | jq -r .Credentials.SecretAccessKey)
+            NON_PROD_AWS_SESSION_TOKEN=$(echo $assumed_role | jq -r .Credentials.SessionToken)
+            export PROFILE_NAME="${TRIGGERING_ACCOUNT_AWS_PROFILE}"
+            aws configure set aws_access_key_id "${NON_PROD_AWS_ACCESS_KEY_ID}" --profile "${PROFILE_NAME}"
+            aws configure set aws_secret_access_key "${NON_PROD_AWS_SECRET_ACCESS_KEY}" --profile "${PROFILE_NAME}"
+            aws configure set aws_session_token "${NON_PROD_AWS_SESSION_TOKEN}" --profile "${PROFILE_NAME}"
+            aws configure set region "eu-west-2" --profile "${PROFILE_NAME}"
+            aws configure set output "json" --profile "${PROFILE_NAME}"
+            cd "${CODEBUILD_SRC_DIR}"
+            platform-helper environment generate --name "${ENVIRONMENT}"
+            platform-helper copilot make-addons
+            AWS_PROFILE="${PROFILE_NAME}" copilot env init --name "${ENVIRONMENT}" --profile "${COPILOT_PROFILE}" --default-config
+            AWS_PROFILE="${PROFILE_NAME}" copilot env deploy --name "${ENVIRONMENT}"
```

Comment:
I think it could probably be refactored so that these and above are the same regardless of the AWS account and just do the role assuming, config etc. if it's a prod account.